### PR TITLE
feat: add artifacts highlight plugin which supports more filters

### PR DESCRIPTION
**Pull request by rootial**
The specific artifacts based on your filters will be highlighted in the map, which can provide more convenience to locate artifacts and transit it to the right place.
image

Enlarge the red dot while planet level <=2 and hollow out the dot for convenience of selecting it.

image

image
image

### CODE REVIEW

**Comment by Stx69** [comment]
working.. nice red dots.

**Review by Stx69 — APPROVED** [review]

File: content/artifacts/easy-highlight-artifacts/plugin.js [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+const ARTIFACT_ANY_TYPE = -1;
+const ARTIFACT_ANY_RARITY = -1;
+
+const PLANET_TYPES = [
+  { value: PLANET_ANY_TYPE, text: "Any" },
+  { value: 0, text: "Planet" },
+  { value: 1, text: "Asteroid Field" },
+  { value: 2, text: "Foundry" },
+  { value: 3, text: "Spacetime Rip" },
+  { value: 4, text: "" },
```
Comment (jacobrosenthal) [inline content/artifacts/easy-highlight-artifacts/plugin.js:23]:
```suggestion
  { value: 4, text: "Quasar" },
```

File: content/artifacts/easy-highlight-artifacts/plugin.js [hunk 2]
```diff
@@ -0,0 +1,426 @@
+import {
+  html,
+  render,
+  useState,
+  useCallback,
+  // @ts-ignore
+} from "https://unpkg.com/htm/preact/standalone.module.js";
+
+const PLANET_LEVELS = [0, 1, 2, 3, 4, 5, 6, 7].map((level) => ({
```
Comment (jacobrosenthal) [inline content/artifacts/easy-highlight-artifacts/plugin.js:9]:
```suggestion
const PLANET_LEVELS = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9].map((level) => ({
```

**Review by jacobrosenthal — COMMENTED** [review]
Besides these nits, the screenshot is a bit large if you can shrink it down more towards like ~20k and it still looks ok.

File: content/artifacts/easy-highlight-artifacts/plugin.js [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+      ctx.fill();
+      ctx.stroke();
+      ctx.closePath();
+    }
+    ctx.restore();
+  }
+
+  async render(container) {
+    render(html`<${App} />`, container);
+  }
```
Comment (phated) [inline content/artifacts/easy-highlight-artifacts/plugin.js:423]: This needs a destroy method that cleans up after preact

File: content/artifacts/easy-highlight-artifacts/plugin.js (hunk 2 above)
Comment (phated) [inline content/artifacts/easy-highlight-artifacts/plugin.js:9]: There are 9 planet levels

File: content/artifacts/easy-highlight-artifacts/plugin.js (hunk 3 above)
Comment (phated) [reply content/artifacts/easy-highlight-artifacts/plugin.js:423]: You track the container as `this.container` and then you use `render(null, this.container)` in the destroy method

Comment (jacobrosenthal) [inline content/artifacts/easy-highlight-artifacts/plugin.js:423]:
```suggestion
  }
  destroy() {
    render(null, this.container);
  }
```

File: content/artifacts/easy-highlight-artifacts/plugin.js [hunk 4]
```diff
@@ -0,0 +1,426 @@
...
+      );
+      ctx.fill();
+      ctx.stroke();
+      ctx.closePath();
+    }
+    ctx.restore();
+  }
+
+  async render(container) {
+    render(html`<${App} />`, container);
```
Comment (jacobrosenthal) [inline content/artifacts/easy-highlight-artifacts/plugin.js:422]:
```suggestion
    this.container = container
    render(html`<${App} />`, container);
```

**Comment by rootial** [comment]
@phated @jacobrosenthal Hi guys, i have addressed comments in latest commit and make some changes to the icon. please see the pic  above. thanks~

**Comment by jacobrosenthal** [comment]
Nice work!

**Comment by jacobrosenthal** [comment]
Are you in discord you should join us #🔌df-plugins channel

**Comment by rootial** [comment]
yeah @jacobrosenthal , i joined discord my account name is JiuKa#7001

**Pull request merged**